Modules/Module_Processing/Module_Population/xASL_stat_GetROIstatistics.m

```diff
@@ -596,22 +608,37 @@
 
 			% Skip PVC for certain ROI and mask combination as the result might not be meaningful
 			bSkipPVC = 0;
-			
+
+            % pGM_here is the main tissue type that is investigated
+            % pWM_here is the other tissue type that is used with PVC
+
+            %   x.S.bTissueMasking - value specifying which subject-wise tissue type we will mask with. Cannot be a vector
+            %                        (loop outside this function for multiple tissue types)
+            % 'GM' = gray matter
+            % 'WM' = white matter
+            % 'WB' = whole brain (= GM+WM)            
+
 			if ~x.S.IsASL
 				pGM_here = ones(size(DataIm)); % no masking
 				bSkipPVC = 1;
-			elseif ~isempty(findstr(lower(namesROIuse{iROI}),'whole brain')) || ~isempty(findstr(lower(namesROIuse{iROI}),'wb')) || ~isempty(findstr(lower(namesROIuse{iROI}),'wholebrain'))
-				pGM_here = pGM+pWM;
-				pWM_here = ones(size(pGM)) - pGM - pWM; % CSF
-				bSkipPVC = 1;
-			elseif  ~isempty(findstr(lower(namesROIuse{iROI}),'white matter')) || ~isempty(findstr(lower(namesROIuse{iROI}),'wm'))  || ~isempty(findstr(lower(namesROIuse{iROI}),'whitematter'))
-				% flip the ROIs:
-				% we are interested in WM instead of pGM
-				pGM_here = pWM;
-				pWM_here = pGM;
-            else % we want GM, so keep tissue masks as they are
-				pGM_here = pGM;
-				pWM_here = pWM;
+            else
+                switch x.S.bTissueMasking

```

Comment:
Didn't we say that bTissueMasking is a boolean? And TissueMasking the vector with strings?

Reply:
No I need to tweak this, one is a global and the other is a local parameter. Should we use gTissueMasking and TissueMasking or something? Or just TissueMasking and TissueMasking_here?